locations/spiders/gov_osservaprezzi_carburanti_it.py

```diff
@@ -0,0 +1,57 @@
+from scrapy import Spider
+from scrapy.http import JsonRequest, Response
+
+from locations.categories import Categories, Fuel, apply_category, apply_yes_no
+from locations.dict_parser import DictParser
+from locations.geo import point_locations
+
+
+class GovOsservaprezziCarburantiITSpider(Spider):
+    name = "gov_osservaprezzi_carburanti_it"
+
+    FUELS = {
+        "Gasolio": Fuel.DIESEL,
+        "GPL": Fuel.LPG,
+        "Metano": Fuel.CNG,
+        # "Blue Diesel": Fuel.ADBLUE, ?
+    }
+
+    BRANDS = {
+        "Api-Ip": {"brand": "IP", "brand_wikidata": "Q646807"},
+        "Esso": {"brand": "Esso", "brand_wikidata": "Q867662"},
+        "Q8": {"brand": "Q8", "brand_wikidata": "Q1634762"},
+        "Tamoil": {"brand": "Tamoil", "brand_wikidata": "Q706793"},
+        "AgipEni": {"brand": "Eni", "brand_wikidata": "Q565594"},
+        "Giap": {"brand": "Giap", "brand_wikidata": None},
+        "Energas": {"brand": "Energas", "brand_wikidata": None},
+        "Keropetrol": {"brand": "Keropetrol", "brand_wikidata": None},
+        "San Marco Petroli": {"brand": "San Marco Petroli", "brand_wikidata": None},
+    }
+
+    def start_requests(self):
+        for lat, lon in point_locations("italy_grid_10km.csv"):
+            yield JsonRequest(
+                "https://carburanti.mise.gov.it/ospzApi/search/zone",
+                data={
+                    "points": [{"lat": lat, "lng": lon}],
+                    "radius": 10,
+                },
+            )
+
+    def parse(self, response: Response):
+        for result in response.json()["results"]:
+            item = DictParser.parse(result)
+
+            apply_category(Categories.FUEL_STATION, item)
+
+            for fuel_name, fuel in self.FUELS.items():
+                apply_yes_no(fuel, item, any(f["name"] == fuel_name for f in result["fuels"]))
```

Comment:
If we flip the loop, we can log failures to improve the mapping. We can also collect the fuel prices, like the cma spider. Something like:

```python
            for fuel in result["fuels"]:
                if fuel["name"] not in self.FUELS:
                    self.crawler.stats.inc_value(
                        "atp/gov_osservaprezzi_carburanti_it/unmapped_fuel/{}".format(fuel["name"])
                    )
                    continue
                key = self.FUELS[fuel["name"]]
                item["extras"]["charge:{}".format(key)] = "{} EUR/1 litre".format(round(fuel["price"], 2))
                apply_yes_no("fuel:{}".format(key), item, True)
```

```python
    FUELS = {
        "Benzina": "petrol",
        "Gasolio": "diesel",
        "GPL": "lpg",
        "Metano": "cng",
        # "Blue Diesel": Fuel.ADBLUE, ?
    }
```

Reply:
Neat, thank you!

For some stations there are 2 different prices for the same fuel depending on self service, is there an established way to distinguish between these? 
![image](https://github.com/alltheplaces/alltheplaces/assets/104327272/b78dd70f-08c5-4c33-a8d8-bb6bcc008815)

Reply:
Technically OSM can do [conditionals](https://wiki.openstreetmap.org/wiki/Conditional_restrictions), but I doubt anything supports using them, and they can be a pain to write. I wouldn't worry about it too much right now, maybe add a TODO comment.